### PR TITLE
946 Fix LB bug in load model and add new LB tests

### LINKED ISSUE #946: Fix bug when running LB with new load models
**Describe the bug**
To reproduce, run `lb_iter` with any load balancer. The phase offset does an invalid lookup in `BaseLB::importProcessorData`.

### PR DESCRIPTION
**Pull request by lifflander**
Fixes #946

### CODE REVIEW

**Comment by lifflander** [comment]
@PhilMiller This needs to be sequenced after #945 and ZoltanLB should be added since that PR fixes some bugs in that.

Also, the test fails because there is a bug in GossipLB. Not sure if its related to load models or not. A bad access in a map it looks like.

File: tests/unit/collection/test_lb.extended.cc [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+INSTANTIATE_TEST_SUITE_P(
+  LoadBalancerExplode, TestLoadBalancer,
+  ::testing::Values(
+    "RandomLB",
+    "RotateLB",
+    "HierarchicalLB",
+    "GossipLB",
+    "GreedyLB"
```
Comment (lifflander) [inline tests/unit/collection/test_lb.extended.cc:125]: TODO: add Zoltan once the #945 is merged which fixes a bug with load modeling and Zoltan.

**Comment by lifflander** [comment]
I'm trying to get to the bottom of the GossipLB bug

**Comment by PhilMiller** [comment]
Would it make sense to open a new issue for the Gossip LB issue, and maybe comment it out of this test, if it's the only one that's failing? It kinda looked like Hierarchical may have been failing too, but the output was kinda jumbled-seeming, possibly due to how the test harness operates.

**Comment by PhilMiller** [comment]
Ok, yeah, the output is a little confusing because of the argument modification and re-initialization. When the vt banner prints out, it's describing the choice of balancer for the test that *just ran*

**Comment by lifflander** [comment]
> Ok, yeah, the output is a little confusing because of the argument modification and re-initialization. When the vt banner prints out, it's describing the choice of balancer for the test that _just ran_

Yes, it's a bit screwy the way I'm setting the LB. We need an interface to configure ArgConfig before startup.

**Comment by PhilMiller** [comment]
> Would it make sense to open a new issue for the Gossip LB issue, and maybe comment it out of this test, if it's the only one that's failing?

I checked that GossipLB is the only failing LB, and opened #948 regarding it. I'd like to move the rest of this in, since it does provide a useful advance for everything that isn't broken anyway.

**Comment by lifflander** [comment]
I agree, let's get this merged.

**Comment by PhilMiller** [comment]
Which of the CI configurations actually runs the extended tests?

**Comment by PhilMiller** [comment]
They weren't an issue for compiling or running with clang or gcc, so maybe they should be on for those?

**Comment by lifflander** [comment]
> Which of the CI configurations actually runs the extended tests?

All the configurations except Intel run extended tests currently. nvcc will be added to that list once we get that added to CI.

**Comment by PhilMiller** [comment]
Looks like they get built, but not run.

**Comment by PhilMiller** [comment]
Or I may be confounding that with whether LB is disabled, which would mask out the relevant test.

I'm going to bed - it's late here.

**Comment by lifflander** [comment]
> Or I may be confounding that with whether LB is disabled, which would mask out the relevant test.
> 
> I'm going to bed - it's late here.

Looks like there's a failure with `RandomLB` on gcc-7. Which ones are skipping it? The only one with LB disabled should be gcc-10.

**Pull request merged**